### PR TITLE
Fix None comparision issue caused by DB errors

### PR DESCRIPTION
**Pull request by sravfeyn**
https://sentry.io/organizations/dimagi/issues/1403426678/?project=1545828&query=server_name%3AMUMGCCWCDPRDPV007&statsPeriod=7d

### CODE REVIEW

File: corehq/sql_db/util.py [hunk 1]
```diff
@@ -365,7 +365,7 @@ def get_standbys_with_acceptible_delay():
 
     return {
         db_ for db_, delay in delays_by_db.items()
-        if 0 <= delay <= acceptable_delays_by_db[db_]
+        if 0 <= delay or -1 <= acceptable_delays_by_db[db_]
```
Comment (snopoke) [inline corehq/sql_db/util.py]: This would need brackets in order to work. It also has a problem when `delay = 0` since `0 or -1 == -1`.

**Review by snopoke — COMMENTED** [review]
Any idea how `delay` ended up being `None` in the first place?

**Comment by sravfeyn** [comment]
> Any idea how delay ended up being None in the first place?

This happened when the SQL statement to get delays fails due to pg errors on those DBs.

File: corehq/sql_db/util.py (hunk 1 above)
Comment (sravfeyn) [reply corehq/sql_db/util.py]: 👍 Will update

**Comment by snopoke** [comment]
> This happened when the SQL statement to get delays fails due to pg errors on those DBs.

Do you mean that plproxy returns None from here `_get_replication_delay_results_from_proxy`? That strikes me as very odd. But if that is the case then a better place to address this would be in that function.

**Review by snopoke — APPROVED** [review]

**Pull request merged**